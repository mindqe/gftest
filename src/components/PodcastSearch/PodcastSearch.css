.podcast-search-main {
  display: flex;
}

.podcast-search-main input {
  width: 120px;
  outline: none;
  border-radius: 5px;
  height: 30px;
  width: 100%;
  border: 1.5px solid #cdc6c6;
}
.podcast-search-badge {
  width: 50px;
  color: white;
  font-size: 12px;
  border-radius: 14px;
  background-color: #2876bf;
  font-weight: 600;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  margin-right: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 30px;
}
