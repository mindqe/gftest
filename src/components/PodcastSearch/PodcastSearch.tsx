import './PodcastSearch.css'

const PodcastSearch = ({filteredPodcasts, searchTerm, setSearchTerm }: {filteredPodcasts: []; searchTerm: string; setSearchTerm: (term: string) => void }) => {
  return (
    <div className="podcast-search-main">        
      <span className="podcast-search-badge">
        {filteredPodcasts.length}
      </span>
      <input
        type="text"
        placeholder="Search podcasts..."
        value={searchTerm}
        onChange={(e) => setSearchTerm(e.target.value)}
      />
    </div>
  );
};

export default PodcastSearch;
