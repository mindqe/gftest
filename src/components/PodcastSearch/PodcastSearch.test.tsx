/**
 * @jest-environment jsdom
*/
import { render, screen, fireEvent } from '@testing-library/react';
import PodcastSearch from './PodcastSearch';

describe('PodcastSearch Component', () => {
  const mockSetSearchTerm = jest.fn();
  const filteredPodcasts = [
    { id: 1, title: 'Podcast One' },
    { id: 2, title: 'Podcast Two' },
  ];
  const searchTerm = 'podcast';

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('renders the component correctly', () => {
    render(
      <PodcastSearch
        filteredPodcasts={filteredPodcasts}
        searchTerm={searchTerm}
        setSearchTerm={mockSetSearchTerm}
      />
    );

    const inputElement = screen.getByPlaceholderText('Search podcasts...');
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveValue(searchTerm);

    const badgeElement = screen.getByText(filteredPodcasts.length.toString());
    expect(badgeElement).toBeInTheDocument();
  });

  describe('displays the correct number of filtered podcasts', () => {
    render(
      <PodcastSearch
        filteredPodcasts={filteredPodcasts}
        searchTerm={searchTerm}
        setSearchTerm={mockSetSearchTerm}
      />
    );

    const badgeElement = screen.getAllByTestId('podcast-badge');
    expect(badgeElement).toHaveTextContent(filteredPodcasts.length.toString());
  });

  describe('calls setSearchTerm with the correct value when typing', () => {
    render(
      <PodcastSearch
        filteredPodcasts={filteredPodcasts}
        searchTerm=""
        setSearchTerm={mockSetSearchTerm}
      />
    );

    const inputElement = screen.getByPlaceholderText('Search podcasts...');
    
    fireEvent.change(inputElement, { target: { value: 'new search' } });
    
    expect(mockSetSearchTerm).toHaveBeenCalledWith('new search');
  });

  describe('matches the snapshot', () => {
    const { container } = render(
      <PodcastSearch
        filteredPodcasts={filteredPodcasts}
        searchTerm={searchTerm}
        setSearchTerm={mockSetSearchTerm}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
