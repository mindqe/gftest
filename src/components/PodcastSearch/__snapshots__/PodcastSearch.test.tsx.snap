// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastSearch Component matches the snapshot 1`] = `
<div>
  <div
    class="podcast-search-main"
  >
    <span
      class="podcast-search-badge"
    >
      2
    </span>
    <input
      placeholder="Search podcasts..."
      type="text"
      value="podcast"
    />
  </div>
</div>
`;
